jaeger-core/src/main/java/com/uber/jaeger/samplers/RemoteControlledSampler.java

```diff
@@ -87,38 +89,35 @@ private void updateSampler() {
       return;
     }
 
-    Sampler newSampler;
-    try {
-      newSampler = extractSampler(response);
-    } catch (UnknownSamplingStrategyException e) {
+    Sampler oldSampler = sampler;
+
+    if (response.getOperationSampling() != null) {
+      OperationSamplingParameters samplingParameters = response.getOperationSampling();
+      if (sampler instanceof PerOperationSampler) {
+        if (((PerOperationSampler) sampler).update(samplingParameters)) {
+          metrics.samplerUpdated.inc(1);
+        }
+      }
+      sampler = new PerOperationSampler(maxOperations, response.getOperationSampling());
+    } else if (response.getProbabilisticSampling() != null) {
+      ProbabilisticSamplingStrategy strategy = response.getProbabilisticSampling();
+      sampler = new ProbabilisticSampler(strategy.getSamplingRate());
+    } else if (response.getRateLimitingSampling() != null) {
+      RateLimitingSamplingStrategy strategy = response.getRateLimitingSampling();
+      sampler = new RateLimitingSampler(strategy.getMaxTracesPerSecond());
+    } else {
       metrics.samplerParsingFailure.inc(1);
-      return; // sampler updateGauge without a new sampler
+      log.info("No strategy present in response.");
     }
 
-    if (!this.sampler.equals(newSampler)) {
-      synchronized (this) {
-        this.sampler = newSampler;
-        metrics.samplerUpdated.inc(1);
-      }
+    if (oldSampler != sampler) {
+      metrics.samplerUpdated.inc(1);
```

Comment:
@yurishkuro This metric updates whenever there is a change in the parameters for a Sampler since the last minute. As such, I think that if we care about it, it should be emitted from a collector. 
We already collect metrics when the sampler update fails, which is far more informative for debugging. What are your thoughts on removing this metric from the client?